src/main/java/seedu/address/model/person/UniqueLogList.java

```diff
@@ -0,0 +1,132 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.List;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.model.person.exceptions.DuplicateLogException;
+import seedu.address.model.person.exceptions.LogNotFoundException;
+
+/**
+ * A list of Logs that enforces the uniqueness between its elements and does not
+ * allow nulls.
+ *
+ * <p>A log is considered unique by comparing {@code Log#isSameLog(Log)}. As such, adding and updating
+ * of Logs uses Log#isSameLog(Log) for equality to ensure that the Log being added or updated is unique
+ * in terms of "identity" in the UniqueLogList.Deleting of Logs uses Log#equals(Object) for equality to ensure
+ * that the Log being deleted is exactly correct.
+ *
+ * <p>Supports a minimal set of list operations.
+ *
+ * @see Log#isSameLog(Log)
+ */
+public class UniqueLogList {
+
+    // observable list allows Java FX listeners to track changes
+    private final ObservableList<Log> internalList = FXCollections.observableArrayList();
+    private final ObservableList<Log> internalUnmodifiableList =
+            FXCollections.unmodifiableObservableList(internalList);
+
+    /**
+     * Returns true if the list contains an equivalent log as the given argument.
+     */
+    public boolean contains(Log log) {
+        requireAllNonNull(log);
+        return this.internalList.stream().anyMatch(log::isSameLog);
+    }
+
+    /**
+     * Returns true if the list contains a log that matches exactly the
+     * given {@code Log}. A stronger form of contains.
+     */
+    public boolean containsExactly(Log log) {
+        requireAllNonNull(log);
+        return this.internalList.stream().anyMatch(log::equals);
+    }
```

Comment:
Do we really need 2 methods for checking if the logs are the same? Since we are defining in our User guide that 2 logs are the same if they have the same title, there should never be 2 logs with the same title but different indexes.

Reply:
The reason for the two methods are to support different notions of equality. We avoid logs with duplicate titles for semantic reasons (e.g. confusing to read, to find), but suppose we want to find exact matches (say, to ensure safe deletion), we might consider using `containsExactly`.

Reply:
I noted that for AB3, they have also have 2 methods for checking if a Person is the same, and I would think that having both methods is good because the first form could be used to ensure uniqueness of the logs and the stronger form could be helpful for testing purposes!